_posts/news/2019-09-18-announcing-scalajs-0.6.29.md

```diff
@@ -0,0 +1,83 @@
+category: news
+tags: [releases]
+permalink: /news/2019/09/18/announcing-scalajs-0.6.29/
+---
+
+
+We are excited to announce the release of Scala.js 0.6.29!
+
+**We highly recommend to all users that they upgrade to this new version.**
+It contains:
+
+* a fix for a major bug in incremental compilation with sbt 1.x, and
+* a major performance improvement, notably for pattern matching.
+
+It also introduces `js.import(moduleName)`, equivalent to JavaScript's dynamic `import()` calls.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to
+[the tutorial]({{ BASE_PATH }}/tutorial/).
+
+## Release notes
+
+If you use `.scala` build files in `project/` and are upgrading from Scala.js 0.6.22 or earlier, do read [the release notes of 0.6.23]({{ BASE_PATH }}/news/2018/05/22/announcing-scalajs-0.6.23/), which contain a source breaking change in that situation.
+
+If upgrading from Scala.js 0.6.14 or earlier, make sure to read [the release notes of 0.6.15]({{ BASE_PATH }}/news/2017/03/21/announcing-scalajs-0.6.15/), which contain important migration information.
+
+As a minor release, 0.6.29 is backward binary compatible with previous releases in the 0.6.x series.
+Libraries compiled with earlier versions can be used with 0.6.29 without change.
+0.6.29 is not forward binary compatible with earlier releases: libraries compiled with 0.6.29 cannot be used by projects using 0.6.{0-28}.
+
+Please report any issues [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Fix for a major bug in incremental compilation with sbt 1.x
+
+When using sbt-scalajs with sbt 1.x, there was a major bug with incremental compilation, causing linker errors ([#3419](https://github.com/scala-js/scala-js/issues/3419) / [#3596](https://github.com/scala-js/scala-js/issues/3596)).
+Scala.js 0.6.29 fixes this issue.
+
+## Major performance improvement notably for pattern matching
+
+Scala.js 0.6.29 features a major performance improvement to all instance tests of the form `x.isInstanceOf[C]` where `C` is a Scala `class`.
+This includes most cases of pattern matching.
+Applications that are pattern matching-intensive can experience speedups of up to 40%.
+
+## Dynamic imports with `js.import()`
+
+ECMAScript 2020 will standardize [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports).
+Scala.js adds support for them with the `js.import[A <: js.Any]()` method, which returns a `js.Promise[A]`.
+The parameter `A` should be a JS trait describing the API of the module, and be given explicitly.
+Since `import` is a keyword in Scala, it must be called with backticks:
+
+{% highlight scala %}
+import scala.scalajs.js
+
+trait FooAPI extends js.Any {
+  def bar(x: Int): Int
+}
+
+val moduleName = "foo.js"
+val promise = js.`import`[FooAPI](moduleName)
+val future = promise.toFuture
+for (module <- future) {
+  println(module.bar(5))
+}
+{% endhighlight %}
+
```

Comment:
Other stuff we should mention somewhere?
- test-interface / test-bridge split (also put it in the artifacts?)
- unicode update
- stack trace suppression of LinkingException
- errors on duplicat exports
- `globalThis` deprecation

Reply:
I don't think the stack trace suppression is worth mentioning in the release notes.

I have added everything else you mention.